overlord/registrystate/registrymgr.go

```diff
@@ -19,13 +19,39 @@
 package registrystate
 
 import (
+	"errors"
 	"fmt"
+	"regexp"
+	"strings"
 
 	"github.com/snapcore/snapd/i18n"
+	"github.com/snapcore/snapd/logger"
 	"github.com/snapcore/snapd/overlord/hookstate"
 	"github.com/snapcore/snapd/overlord/state"
 )
 
+type RegistryManager struct{}
+
+func Manager(st *state.State, hookMgr *hookstate.HookManager, _ *state.TaskRunner) *RegistryManager {
```

Comment:
this needs at least a happy path test

Reply:
I didn't add it because the next PR (the one already open) tests this implicitly. I'll add add a small check